lib/improver/wind_direction.py

```diff
@@ -320,47 +327,62 @@ def calc_confidence_measure(self):
            is essentially meaningless.
            We therefore substitute this meaningless average wind
-           direction value for the wind direction taken from the first
-           ensemble realization.
+           direction value for the wind direction calculated from a larger
+           sample by smoothing across a neighbourhood of points before
+           rerunning the main technique.
+           This is invoked rarely (1 : 100 000)
 
         Arguments:
             where_low_r (np.array):
                 Array of boolean values. True where original wind direction
                 estimate has low confidence. These points are replaced
                 according to self.backup_method
-            first_member (np.array):
-                Array of wind direction data from the first ensemble
-                realization
+            wdir_cube (iris.cube.Cube):
+                Contains array of wind direction data (realization, y, x)
 
         Uses:
-            self.wdir_slice.data (np.ndarray):
-                3D array - wind direction angles from ensembles (in degrees).
-            self.wdir_slice_mean.data (np.ndarray):
-                2D array - average wind direction angle (in degrees).
+            self.wdir_slice_mean (iris.cube.Cube):
+                Containing average wind direction angle (in degrees).
+            self.wdir_complex (np.ndarray):
+                3D array - wind direction angles from ensembles (in complex).
             self.r_vals_slice.data (np.ndarray):
                 2D array - Radius taken from average complex wind direction
                 angle.
             self.r_thresh (float):
                 Any r value below threshold is regarded as meaningless.
+            self.realization_axis (int):
+                Axis to collapse over.
+            self.n_realizations (int):
+                Number of realizations available in the plugin. Used to set the
+                neighbourhood radius as this is used to adjust the radius again
+                in the neighbourhooding plugin.
 
         Defines:
             self.wdir_slice_mean.data (np.ndarray):
                 2D array - Wind direction degrees where ambigious values have
                 been replaced with data from first ensemble realization.
         """
-
         if self.backup_method == 'neighbourhood':
-            improved_values = np.full_like(self.wdir_slice_mean.data, None)
+            # Performs smoothing over a 6km square neighbourhood.
+            # Then calculates the mean wind direction.
+            child_class = WindDirection(backup_method="first_realization")
+            child_class.wdir_complex = self.nbhood.process(
+                wdir_cube.copy(data=self.wdir_complex)).data
+            child_class.realization_axis = self.realization_axis
+            child_class.wdir_slice_mean = self.wdir_slice_mean.copy()
+            child_class.wind_dir_mean()
```

Comment:
Not necessarily caused by this change, but it might help clarity if the `wind_dir_mean` method was renamed to e.g. `calc_wind_dir_mean`, as `wind_dir_mean` could easily be confused with a variable name.